runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -1530,6 +1540,112 @@ J9::CodeGenerator::lowerTreeIfNeeded(
 
    }
 
+void
+J9::CodeGenerator::runGlobalLiveVariablesForGC()
+   {
+   auto comp = self()->comp();
+   auto trHeapMemory = self()->trHeapMemory();
+   auto trMemory = self()->trMemory();
+
+   if (TR::Compiler->om.areValueTypesEnabled() &&
+       (comp->getOption(TR_EnableParanoidOptCheck) || debug("paranoidOptCheck")))
+      comp->verifyCFG(comp->getMethodSymbol());
+
+   if (!comp->getFlowGraph()->getStructure())
+      {
+      comp->getFlowGraph()->setStructure(TR_RegionAnalysis::getRegions(comp));
```

Comment:
Region analysis is expensive, we should only do it if we're to run liveness analysis